### PR TITLE
SISRP-26457 - Student Success Card need to show appropriate message instead of blank details

### PR DESCRIPTION
**Pull request by davidlimcheng**
https://jira.berkeley.edu/browse/SISRP-26457

* Some friendly UI changes for unexpected values.
* Also in this PR:  nests the `isLoading = false` flag into a `.finally()` promise so we don't get infinitely spinning surprises.
* QA PR to come upon merge.

### CODE REVIEW

File: src/assets/templates/widgets/student_success/student_success.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <h2>Student Success</h2>
   </div>
   <div data-cc-spinner-directive="studentSuccess.isLoading">
-    <div data-ng-if="studentSuccess.termGpa.length" data-ng-include="'widgets/student_success/gpa_trends.html'"></div>
+    <div data-ng-include="'widgets/student_success/gpa_trends.html'"></div>
```
Comment (johncrossman) [inline src/assets/templates/widgets/student_success/student_success.html]: @davidlimcheng advisors will see "GPA not yet available" for all students with nil `studentSuccess.termGpa`.  At the risk of sounding nit-picky, will the phrase `not yet available` always be accurate?  I.e., will we have cases in which grades never arrive?

File: src/assets/templates/widgets/student_success/units.html [hunk 2]
```diff
@@ -10,27 +10,31 @@
...
+          <span data-ng-if="!gpaUnits.totalUnitsNotForGpa">-</span>
+          <span data-ng-if="gpaUnits.totalUnitsNotForGpa" data-ng-bind="gpaUnits.totalUnitsNotForGpa | number:1"></span>
         </td>
       </tr>
       <tr>
         <td>Transfer Units</td>
         <td class="cc-table-right" colspan="2">
-          <span data-ng-bind="gpaUnits.transferUnitsAccepted | number:1"></span>
+          <span data-ng-if="!gpaUnits.transferUnitsAccepted">-</span>
+          <span data-ng-if="gpaUnits.transferUnitsAccepted" data-ng-bind="gpaUnits.transferUnitsAccepted | number:1"></span>
```
Comment (johncrossman) [inline src/assets/templates/widgets/student_success/units.html]: @davidlimcheng if a student has zero transfer units then it's likely s/he never will (whilst a student at UCB). I suggest we ask Flint if the `<tr>` above should be hidden when `!gpaUnits.transferUnitsAccepted`?

File: src/assets/templates/widgets/student_success/units.html [hunk 3]
```diff
@@ -10,27 +10,31 @@
...
+          <span data-ng-if="!gpaUnits.totalUnitsForGpa">-</span>
+          <span data-ng-if="gpaUnits.totalUnitsForGpa" data-ng-bind="gpaUnits.totalUnitsForGpa | number:1"></span>
         </td>
       </tr>
       <tr>
         <td>Units not for GPA</td>
         <td class="cc-table-right" colspan="2">
-          <span data-ng-bind="gpaUnits.totalUnitsNotForGpa | number:1"></span>
+          <span data-ng-if="!gpaUnits.totalUnitsNotForGpa">-</span>
+          <span data-ng-if="gpaUnits.totalUnitsNotForGpa" data-ng-bind="gpaUnits.totalUnitsNotForGpa | number:1"></span>
```
Comment (johncrossman) [inline src/assets/templates/widgets/student_success/units.html]: Then again, sometimes "zero items" is useful info.  Another question for Flint.

File: src/assets/templates/widgets/student_success/student_success.html (hunk 1 above)
Comment (davidlimcheng) [inline src/assets/templates/widgets/student_success/student_success.html]: @johncrossman Yes, definitely that's a great point - for example, I believe a LAW student with no other history will never have grades arrive.  I'll discuss this with Flint and make any necessary changes.

File: src/assets/templates/widgets/student_success/units.html (hunk 3 above)
Comment (davidlimcheng) [inline src/assets/templates/widgets/student_success/units.html]: @johncrossman I could not agree more.  "Less is more", will discuss with Flint.

File: src/assets/templates/widgets/student_success/units.html (hunk 2 above)
Comment (davidlimcheng) [inline src/assets/templates/widgets/student_success/units.html]: Will do.

**Comment by davidlimcheng** [comment]
Per discussion with Flint - 
* Nix GPA section if no GPA to show.
* Nix TransferUnits of no transfer units to show.
* Nix UnitsNotForGpa when none to show.

First Amendment to come.

**Comment by johncrossman** [comment]
I support the First Amendment – let me know when it's all in, @davidlimcheng. The Second Amendment... Hmm.

**Comment by davidlimcheng** [comment]
Thanks @johncrossman, should be good to go now.

**Comment by johncrossman** [comment]
+1 @davidlimcheng

**Pull request merged**